Userland/Games/Chess/ChessWidget.cpp

```diff
@@ -536,91 +538,191 @@ ErrorOr<String> ChessWidget::get_fen() const
     return TRY(m_playback ? m_board_playback.to_fen() : m_board.to_fen());
 }
 
-ErrorOr<void> ChessWidget::import_pgn(Core::File& file)
+ErrorOr<void, PGNParseError> ChessWidget::import_pgn(Core::File& file)
 {
-    m_board = Chess::Board();
+    reset();
+    enum TokenType {
+        MOVE,
+        TAG_SYMBOL,
+        TAG_STRING,
+        COMMENT,
+        GAME_TERMINATOR,
+        RAV_START,
+        RAV_END,
+        NAG
+    };
```

Comment:
I forgot to say previously, we prefer to use `enum class`, which among other things, properly scopes the enum values. Also, name them `LikeThis` not `LIKE_THIS`.